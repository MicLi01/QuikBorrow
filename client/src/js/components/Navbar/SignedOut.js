import React from 'react';
import { Link } from 'react-router-dom';
import { Toolbar } from '@material-ui/core';
import styled from 'styled-components'

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
    margin-left: 20px;
`

const SignedOut = () => {
    return (
        <Toolbar>
            <UnstyledLink to='/login'>Login</UnstyledLink>
            <UnstyledLink to='/register'>Register</UnstyledLink>
        </Toolbar>
    )
}

export default SignedOut;