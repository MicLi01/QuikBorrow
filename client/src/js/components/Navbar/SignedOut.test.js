import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from "react-router-dom";
import SignedOut from './SignedOut';

Enzyme.configure({ adapter: new Adapter() });

describe('<SignedOut />', () => {
    /** @type {Enzyme.ReactWrapper} */
    let wrapper

    let element

    beforeEach(() => {  
        wrapper = shallow(<SignedOut />);
        element = selector => wrapper.find(selector);
    });

    it('should render', () => {
		  const component = renderer.create(
        <MemoryRouter>
          <SignedOut />
        </MemoryRouter>);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have a login link', () => {
      expect(element('#login')).toHaveLength(1);
    });

    it('should have a register link', () => {
      expect(element('#register')).toHaveLength(1);
    });
    
});