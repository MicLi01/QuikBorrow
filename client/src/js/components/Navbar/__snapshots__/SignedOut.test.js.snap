// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignedOut /> should render 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
>
  <a
    className="sc-bdVaJa keFmjn"
    href="/login"
    id="login"
    onClick={[Function]}
  >
    Login
  </a>
  <a
    className="sc-bdVaJa keFmjn"
    href="/register"
    id="register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
